vspace/src/main/java/edu/asu/diging/vspace/web/staff/StaffSearchController.java

```diff
@@ -0,0 +1,79 @@
+package edu.asu.diging.vspace.web.staff;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.util.StringUtils;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import edu.asu.diging.vspace.core.model.IVSpaceElement;
+import edu.asu.diging.vspace.core.model.impl.ExternalLinkValue;
+import edu.asu.diging.vspace.core.services.IStaffSearchManager;
+
+@Controller
+public class StaffSearchController {
+
+    @Autowired
+    private IStaffSearchManager staffSearchManager;
+
+    @Value("${page_size}")
+    private int pageSize;
+
+    @RequestMapping(value = "/staff/search")
+    @GetMapping
+    public String getAllSearchedElementsWithoutNum(@RequestParam(name = "searchText") String searchString) {
+        return String.format("redirect:/staff/search/1?searchText=%s", searchString);
+    }
+
+    @RequestMapping(value = "/staff/search/{page}")
+    @GetMapping
+    public String getAllSearchedElements(@PathVariable(required = false) String page, Model model,
+            @RequestParam(name = "searchText") String searchString) {
+        HashSet<IVSpaceElement> elementSet = staffSearchManager.getAllSearchedElements(searchString);
+        List<IVSpaceElement> elementList = new ArrayList<>(elementSet);
+        int pageNo;
+        page = StringUtils.isEmpty(page) ? "1" : page;
+        try {
+            pageNo = validatePageNumber(Integer.parseInt(page), elementList.size());
+        } catch (NumberFormatException numberFormatException) {
+            pageNo = 1;
+        }
+        int startIndex = (pageNo - 1) * pageSize;
+        int endIndex = startIndex + pageSize;
+        if (endIndex < elementSet.size()) {
+            elementList = elementList.subList(startIndex, endIndex);
+        } else {
+            if (startIndex < elementSet.size()) {
+                elementList = elementList.subList(startIndex, elementSet.size());
+            }
+        }
```

Comment:
this logic won't work very well for two reasons:
- all search results will be loaded into memory before the list is cut down to only one page
- the results will now always first show all spaces, then all modules, ...

Eventually, we will want to integrate elasticsearch. For now, let's do it this way:
- on the front end, show four tabs or something similar (talk to adhvik) one for each type of elements
- then search with pagination for each type
- show each type in one of the tabs (each with their own pagintation), so the user can choose what type they want to see (e.g. see all found spaces or all found modules)